src/app/component/shared/detail-view/view/view.component.ts

```diff
@@ -0,0 +1,107 @@
+import { Component, Input, OnInit } from "@angular/core";
+import { DateTime, Duration } from "luxon";
+import { toRelative } from "src/app/interfaces/apiInterfaces";
+
+@Component({
+  selector: "app-view",
+  template: `
+    <ng-container *ngIf="!children; else hasChildren">
+      <pre *ngIf="styling === CodeStyling.Code">{{ display }}</pre>
+      <p *ngIf="styling === CodeStyling.Plain">{{ display }}</p>
+      <app-checkbox
+        *ngIf="styling === CodeStyling.Checkbox"
+        [checked]="display"
+        [disabled]="true"
+        [ngClass]="{}"
+      ></app-checkbox>
+    </ng-container>
+    <ng-template #hasChildren>
+      <app-view *ngFor="let child of children" [view]="child"></app-view>
+    </ng-template>
+  `,
+})
+export class ViewComponent implements OnInit {
+  @Input() view: ModelView;
+  public display: string | number | boolean;
+  public children: ModelView[];
+  public styling: CodeStyling = CodeStyling.Plain;
+  public CodeStyling = CodeStyling;
+  private errorText = "(error)";
+  private loadingText = "(loading)";
+  private noValueText = "(no value)";
+
+  constructor() {}
+
+  ngOnInit(): void {
+    this.humanize(this.view);
+  }
+
+  private humanize(value: any) {
+    if (value === null || value === undefined) {
+      this.display = this.noValueText;
+    } else if (value instanceof DateTime) {
+      this.display = `${value.toISO()} (${value.toRelative()})`;
+    } else if (value instanceof Duration) {
+      this.display = `${value.toISO()} (${toRelative(value)})`;
+    } else if (value instanceof Array) {
+      this.children = value;
+    } else if (value instanceof Blob) {
+      this.humanizeBlob(value);
+    } else if (typeof value === "object") {
+      this.humanizeObject(value);
+    } else if (typeof value === "boolean") {
+      this.styling = CodeStyling.Checkbox;
+      this.display = value;
+    } else {
+      this.display = value.toString();
+    }
+  }
+
+  /**
+   * Convert object to human readable output
+   * @param answer Answer output
+   */
+  private humanizeObject(value: object) {
+    this.setLoading();
+
+    try {
+      this.styling = CodeStyling.Code;
+      this.display = JSON.stringify(value);
```

Comment:
I'd like to use a tree view here to expose an arbitrary object graph.  You can even reuse this `view` component for the nodes!

This can be done later on.

Reply:
Oh and that doesn't negate the utility of `CodeStyling.Code` - there are some code/script block we'll want to render as 'code'

Reply:
https://github.com/QutEcoacoustics/workbench-client/issues/183